ft_statistics_mvpa.m

```diff
@@ -171,257 +173,132 @@
-    cfg.connectivity = true(1);
-  elseif isfield(cfg, 'channel')
-    cfg.neighbours   = ft_getopt(cfg, 'neighbours', []);
-    cfg.connectivity = channelconnectivity(cfg);
-  else
-    % there is no connectivity in the spatial dimension
-    cfg.connectivity = false(size(dat,1));
-  end
+    cfg.mvpa.metric     = ft_getopt(cfg.mvpa, 'metric', 'accuracy');
 else
-  % use the specified connectivity: op hoop van zegen
-end
-
-if isfield(cfg, 'dim') && isfield(cfg, 'dimord') && contains(cfg.dimord, 'time') && ~isempty(cfg.mvpa.timwin)
-  % create neighourhood matrix for time dimension
-  dimtok = tokenize(cfg.dimord, '_');
-  timdim = find(strcmp(dimtok, 'time'));
-  T = ones(cfg.dim(timdim));
-  T = T - triu(T, floor(cfg.mvpa.timwin./2)) - tril(T, -ceil(cfg.mvpa.timwin./2)) > 0;
-  if isfield(cfg.mvpa, 'time')&&~isempty(cfg.mvpa.time)
-    % this cannot be dealt with by ft_getopt, because the lower level code
-    % will in that case not properly set the default value
-    T = T(cfg.mvpa.time,:);
-  end
-else
-  T = [];
+    cfg.mvpa.metric     = ft_getopt(cfg.mvpa, 'metric', 'mae');
 end
+cfg.mvpa.feedback   = ft_getopt(cfg.mvpa, 'feedback',   'yes');
 
-if ~isempty(T) && ~istrue(cfg.searchlight)
-  ft_warning('specifying cfg.searchlight = ''yes'' since you want a moving window mvpa');
-  cfg.searchlight = 'yes';
+if isfield(cfg, 'dimord')
+    cfg.mvpa.dimension_names = ft_getopt(cfg.mvpa, 'dimension_names', [{'samples'} tokenize(cfg.dimord, '_')]);
 end
 
 % flip dimensions such that the number of trials comes first
 dat = dat.';
 
-% if cfg.dim has two entries which are non-singleton then the data has been
-% 3D before. We must reshape it from 2D to 3D to run it in MVPA-Light
-data_is_3D = (numel(cfg.dim) > 1 && all(cfg.dim > 1)) && istrue(cfg.reshape3D);  % checks whether data was 3D before being reshaped
-if data_is_3D
-  if isfield(cfg, 'dimord')
-    dimtok = tokenize(cfg.dimord, '_');
-    cfg.mvpa.dimension_names = {'samples' dimtok{1} dimtok{2}};
-  end
-  dat = reshape(dat, size(dat,1), cfg.dim(1), cfg.dim(2));
+if isfield(cfg, 'dim')
+    % MVPA-Light expects the original multi-dimensional array
+    dat = reshape(dat, [size(dat,1) cfg.dim]);
 end
 
-y = design;
+%% backward compatibility
+cfg.mvpa = ft_checkconfig(cfg.mvpa, 'renamed', {'param', 'hyperparameter'});
+ft_checkconfig(cfg, 'deprecated', {'timextime' 'searchlight'});
+ft_checkconfig(cfg.mvpa, 'deprecated', {'balance' 'normalise' 'replace'});
 
-%% perform sanity checks on parameters
-if istrue(cfg.timextime) && istrue(cfg.searchlight)
-  ft_error('you should not set timextime = ''yes'' and searchlight = ''yes'' simultaneously')
+if isfield(cfg,'timextime') && strcmp(cfg.timextime, 'yes')
+    cfg.generalize = 'time';
+end
+if isfield(cfg,'searchlight') && strcmp(cfg.searchlight, 'yes')
+    cfg.features = 3;
+end
+if isfield(cfg,'normalise') && ~isfield(cfg.mvpa,'preprocess')
+    add_to_preprocess(cfg.normalise);
```

Comment:
our convention is to have the subfunctions have explicit output arguments. I am not familiar with this syntax, but I assume that it somehow ends up in the cfg of the caller function anyhow? 

to avoid confusion, I suggest to change this (i.e. have add_to_preprocess output cfg.preprocess explicitly.)

Reply:
It's an internal function so it has access to everything in the caller. I don't mind changing the syntax though for consistency.